config/v1alpha1/types_cluster_monitoring.go

```diff
@@ -117,7 +117,7 @@ const (
 // alertmanagerConfig provides configuration options for the default Alertmanager instance
 // that runs in the `openshift-monitoring` namespace. Use this configuration to control
 // whether the default Alertmanager is deployed, how it logs, and how its pods are scheduled.
-// +kubebuilder:validation:XValidation:rule="self.deploymentMode == 'CustomConfig' ? has(self.customConfig) && !has(self.customConfig) : true",message="customConfig is required when deploymentMode is CustomConfig, and forbidden otherwise"
+// +kubebuilder:validation:XValidation:rule="(self.deploymentMode == 'CustomConfig' && has(self.customConfig)) || (self.deploymentMode != 'CustomConfig' && !has(self.customConfig))",message="customConfig is required when deploymentMode is CustomConfig, and forbidden otherwise"
```

Comment:
This should be as below, we always use ternaries for CEL discrimination, please be consistent
```suggestion
// +kubebuilder:validation:XValidation:rule="self.deploymentMode == 'CustomConfig' ? has(self.customConfig)) : !has(self.customConfig))",message="customConfig is required when deploymentMode is CustomConfig, and forbidden otherwise"
```